includes/admin/discounts/discount-actions.php

```diff
@@ -108,8 +108,8 @@ function edd_admin_add_discount( $data = array() ) {
 	}
 
 	// Meta values.
-	$to_add['product_reqs']      = isset( $data['product_reqs']      ) ? wp_parse_id_list( $data['product_reqs']      ) : '';
-	$to_add['excluded_products'] = isset( $data['excluded_products'] ) ? wp_parse_id_list( $data['excluded_products'] ) : '';
+	$to_add['product_reqs']      = isset( $data['product_reqs'] ) ? array_unique( array_map( 'esc_attr', $data['product_reqs'] ) ) : '';
```

Comment:
```suggestion
	$to_add['product_reqs']      = isset( $data['product_reqs'] ) ? array_unique( array_map( 'sanitize_text_field', $data['product_reqs'] ) ) : '';
```

That feels like a better choice since we're saving data and not displaying it.

Will need updating in all other areas.